pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2070,13 +2072,14 @@ private Set<String> getAllInstancesForTable(String tableNameWithType) {
   }
 
   /**
-   * Returns the ZK metdata for the given jobId
+   * Returns the ZK metdata for the given jobId and jobType
    * @param jobId the id of the job
+   * @param jobType Job Path
    * @return Map representing the job's ZK properties
    */
   @Nullable
-  public Map<String, String> getControllerJobZKMetadata(String jobId) {
-    String controllerJobResourcePath = ZKMetadataProvider.constructPropertyStorePathForControllerJob();
+  public Map<String, String> getControllerJobZKMetadataForJobType(String jobId, ControllerJobType jobType) {
```

Comment:
(minor) Suggest keeping the existing method name
```suggestion
  public Map<String, String> getControllerJobZKMetadata(String jobId, ControllerJobType jobType) {
```